### PR TITLE
[API] MoveConverter::try_into_vm_value() special-cases ASCII::String

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

So that a string can be passed in as a table key when querying (new API coming up in https://github.com/aptos-labs/aptos-core/pull/393)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Y
## Test Plan
unittest

### CODE REVIEW

**Review by davidiw — COMMENTED** [review]
forgive my naivete, this seems counter intuitive. an ASCII::String is not a primitive, it is a struct containing a vec of bytes. or is this due to the syntactic sugar of "hello" -- the quotes and ascii characters?

**Comment by msmouse** [comment]
Yes this is to be compatible with the existing API behavior where a ASCII::String is special cased when being returned to the client (as a string instead of a vector).

File: api/types/src/convert.rs [hunk 1]
```diff
@@ -498,14 +498,21 @@ impl<'a, R: MoveResolver + ?Sized> MoveConverter<'a, R> {
...
-                "Expecting `MoveStructLayout::WithFields, getting {:?}",
-                layout
-            );
-        };
+        let (struct_tag, field_layouts) =
+            if let MoveStructLayout::WithTypes { type_, fields } = layout {
+                (type_, fields)
+            } else {
+                bail!(
+                    "Expecting `MoveStructLayout::WithFields, getting {:?}",
```
Comment (msmouse) [inline api/types/src/convert.rs:506]: fix: WithTypes

**Review by zekun000 — APPROVED** [review]

**Pull request closed without merging**